src/api/topicsApi.ts

```diff
@@ -1,19 +1,25 @@
 import axios from "axios";
 import { message } from "antd";
+import { ActionTypes } from "../store/types";
 
 const request = axios.create({
     withCredentials: true,
     baseURL: `/api/topics`,
 });
 
-const getTopics = (topicName, t) => {
+const getTopics = (dispatch, topicName, t) => {
```

Comment:
suggestions: if you're going to pass optional parameters, pass an object as parameter and do the following:
```
const getTopics = ({ dispatch, topicName, t }) => {
```

Reply:
Also, please create a type interface for this function